barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -454,47 +445,35 @@ std::optional<index_t> ContentAddressedCachedTreeStore<LeafValueType>::find_leaf
     ReadTransaction& tx,
     bool includeUncommitted) const
 {
-    Indices committed;
-    std::optional<index_t> result = std::nullopt;
-    FrKeyType key = leaf;
-    std::vector<uint8_t> value;
-    bool success = dataStore_->read_leaf_indices(key, committed, tx);
-    if (success) {
-        index_t sizeLimit = constrain_tree_size(requestContext, tx);
-        if (!committed.indices.empty()) {
-            for (index_t ind : committed.indices) {
-                if (ind < start_index) {
-                    continue;
-                }
-                if (ind >= sizeLimit) {
-                    continue;
-                }
-                if (!result.has_value()) {
-                    result = ind;
-                    continue;
-                }
-                result = std::min(ind, result.value());
-            }
-        }
-    }
     if (includeUncommitted) {
         // Accessing indices_ under a lock
         std::unique_lock lock(mtx_);
         auto it = indices_.find(uint256_t(leaf));
-        if (it != indices_.end() && !it->second.indices.empty()) {
-            for (index_t ind : it->second.indices) {
-                if (ind < start_index) {
-                    continue;
-                }
-                if (!result.has_value()) {
-                    result = ind;
-                    continue;
-                }
-                result = std::min(ind, result.value());
+        if (it != indices_.end()) {
+            // we have an uncommitted value, we will return from here
+            if (it->second >= start_index) {
+                // we have a qualifying value
+                return std::make_optional(it->second);
             }
+            return std::nullopt;
```

Comment:
Just to confirm: in this branch, since we know the data is in uncommitted (but it's outside the requested range), we know it cannot be in committed, so we can return early and don't need to look for it in the datastore, right?

Reply:
Yes. Leaves never change index, so if it's found in uncommitted it won't be at a different index in committed if it is there at all.